firmware/components/driver_sndmixer/sndmixer.c

```diff
@@ -241,8 +241,8 @@ static void sndmixer_task(void *arg) {
                 (channel[ch].chunksz << 16);  // reset dds acc; we have parsed chunksize samples.
             channel[ch].chunksz = r;          // save new chunksize
           }
-          // Multiply by volume, add to cumulative sample
-          s += (int32_t)(channel[ch].buffer[channel[ch].dds_acc >> 16]) * channel[ch].volume;
+          // Multiply by volume, add to cumulative sample. Limit volume to 1/2 maximum.
+          s += (int32_t)(channel[ch].buffer[channel[ch].dds_acc >> 16]) * channel[ch].volume / 2;
```

Comment:
Why was this done? I had not noticed it before

Reply:
For quick "fixing" of their "it's too loud" issue, I missed this during review. We should undo that change and/or add an option for supplying a max volume reduction factor:

Instead of /2 one could write *(CONFIG_VOLUME_LIMIT/100)

Reply:
Not "too loud" but "overflows". We didn't figure out where the overflow was coming from, but halving is exactly what is needed to limit to 100% volume.

Doing the `*(CONFIG_VOLUME_LIMIT/100)` would usually multiply with 0, because integers...

Reply:
Interesting, because for CZ20 we're now seeing exactly 50% of peak volume. Removing this division results in a perfect 5V peak-to-peak sinusoidal (which is exactly what we expect) without clipping.